src/Domain/JapaneseParser.php

```diff
@@ -97,4 +97,29 @@ public function getParsedTokens(string $text, Language $lang) {
         return $tokens;
     }
 
+    public function getReading(string $text, Language $lang) {
+        // Japanese is an exception
+        $mecab_file = tempnam(sys_get_temp_dir(), "mecab_to_db");
+        $mecab_args = ' -O yomi ';
+        if (file_exists($mecab_file)) { 
+            unlink($mecab_file); 
+        }
+        $fp = fopen($mecab_file, 'w');
+        fwrite($fp, $text . "\n");
+        fclose($fp);
+        $mecab = JapaneseParser::MeCab_command($mecab_args);
+        $handle = popen($mecab . $mecab_file, "r");
+        // $mecab_str Output string
+        $mecab_str = '';
+        while (($line = fgets($handle, 4096)) !== false) {
```

Comment:
What's 4096?

Reply:
It stands for "read 4096 bytes from the file" ([function.fgets.php](https://www.php.net/manual/en/function.fgets.php)). I don't like to read file without size limit, as it can quickly paralyze the server. This value is arbitrary, I can change or delete it if you prefer.

Reply:
I'd just file_get_contents(), it's going to be pretty short, isn't it?  The call to mecab is going to be the performance hit, not a file read.  If it turns out to be slow later, it can be revised, but it won't. :-)

Reply:
Well, I was not very satisfied by these chunks of code on LWT-fork as well, so I was motivated to improve things. It appears I changed most of the function, things are much better now.